### PR TITLE
[W5.11][T15-A1] Chan Jin Jia

### PR DESCRIPTION
**Pull request by jaronchan**
[LO-2KLoC] Added Sort Command with updated tests

### CODE REVIEW

**Comment by jaronchan** [comment]
I am not sure what has caused the Travis CI build error -- details given is null

File: src/seedu/addressbook/comparators/PersonNameComparator.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.Comparator;
+
+/**
+ *
+ * Compares names of Persons.
+ *
+ */
+public class PersonNameComparator implements Comparator<Person>{
+    public int compare(Person p1, Person p2) {
+        return p1.getName().fullName.compareTo(p2.getName().fullName);
```
Comment (madsonic) [inline src/seedu/addressbook/comparators/PersonNameComparator.java:14]: is it a good idea to use `fullName` to compare?

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 2]
```diff
@@ -122,6 +123,14 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
         }
     }
 
+    /**
+     * Sorts the persons list in alphabetical order.
+     */
+    public void sort() {
+        PersonNameComparator personNameComparator = new PersonNameComparator();
+        Collections.sort(internalList, personNameComparator);
```
Comment (madsonic) [inline src/seedu/addressbook/data/person/UniquePersonList.java:144]: Lambdas would be appropriate here

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (madsonic) [inline src/seedu/addressbook/parser/Parser.java]: you are encouraged to use explicit imports

**Comment by madsonic** [comment]
Remember that users might not find out about this new command if it is not documented properly

I re-ran Travis for your PR

Because you did not satisfactorily complete 1kLoC LO, you are required to 2 enhancement if you want to recover lost marks. You can work until the next tutorial to add in the additional enhancement. Do remember to update user guide, test, etc.

**Comment by jaronchan** [comment]
@madsonic Hi there, but I did complete my 1kLoC and tested it with you during tutorial! I implemented a command that allows the editing of contact details. Please advice!

File: doc/UserGuide.md [hunk 4]
```diff
@@ -49,6 +49,25 @@ Examples:
 * `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
 * `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
 
+### Editing an existing person: `edit`
```
Comment (madsonic) [inline doc/UserGuide.md:52]: guide is very clear 👍

File: src/seedu/addressbook/commands/EditCommand.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static seedu.addressbook.ui.TextUi.DISPLAYED_INDEX_OFFSET;
+
+public class EditCommand extends Command{
```
Comment (madsonic) [inline src/seedu/addressbook/commands/EditCommand.java:14]: coding style violation. [read this](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html)

File: src/seedu/addressbook/commands/EditCommand.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(int targetVisibleIndex , String name,
+            String phone, boolean isPhonePrivate,
+            String email, boolean isEmailPrivate,
+            String address, boolean isAddressPrivate,
+            Set<String> tags) throws IllegalValueException {
+        super(targetVisibleIndex);
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
```
Comment (madsonic) [inline src/seedu/addressbook/commands/EditCommand.java:45]: you can make use of `addAll`

File: test/expected.txt [hunk 7]
```diff
@@ -328,6 +332,85 @@
...
+|| 3 persons listed!
+|| ===================================================
+|| Enter command: || [Command entered:  find Adam Dickson]
+|| 	1. Adam Brown Phone: 555555 Email: newadam@gmail.com Address: 555, omega street Tags: [new]
+|| 	2. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 
+|| 2 persons listed!
+|| ===================================================
+|| Enter command: || [Command entered:  edit 2 Dickson Ee p/666666 pe/dickson@nus.edu.sg pa/444, delta street t/friends]
+|| Edited Person: Dickson Ee Phone: 666666 Email: (private) dickson@nus.edu.sg Address: (private) 444, delta street Tags: [friends]
```
Comment (madsonic) [inline test/expected.txt:405]: private details shown when using `list`. inconsistent with [userguide](https://github.com/nus-cs2103-AY1718S2/addressbook-level2/pull/693/commits/43dfd079a71108146bb7ec75ff76a45e20497f92#diff-089f3d23623ca3d148d66f89ad325a42R63)

**Comment by madsonic** [comment]
good job! 
Close this PR if you have no other questions

**Pull request closed without merging**